logging/src/main/java/org/jboss/as/logging/formatters/StructuredFormatterResourceDefinition.java

```diff
@@ -0,0 +1,462 @@
+                if (keyOverrides == null) {
+                    configuration = logContextConfiguration.addFormatterConfiguration(null, className, name);
+                } else {
+                    configuration = logContextConfiguration.addFormatterConfiguration(null, className, name, "keyOverrides");
+                    configuration.setPropertyValueString("keyOverrides", keyOverrides);
+                }
+            } else if (changeProperty(configuration, "keyOverrides", keyOverrides)) {
+                LoggingLogger.ROOT_LOGGER.tracef("Removing then adding formatter '%s' at '%s'", name, LoggingOperations.getAddress(operation));
+                logContextConfiguration.removeFormatterConfiguration(name);
+                configuration = logContextConfiguration.addFormatterConfiguration(null, className, name, "keyOverrides");
+                configuration.setPropertyValueString("keyOverrides", keyOverrides);
+            }
+
+            // Process the attributes
+            for (AttributeDefinition attribute : ATTRIBUTES) {
+                if (attribute == META_DATA) {
+                    final String metaData = modelValueToMetaData(META_DATA.resolveModelAttribute(context, model));
+                    if (metaData != null) {
+                        if (changeProperty(configuration, "metaData", metaData)) {
+                            configuration.setPropertyValueString("metaData", metaData);
+                        }
+                    } else {
+                        configuration.removeProperty("metaData");
+                    }
+                } else if (attribute == KEY_OVERRIDES) {
+                    // Ignore the key-overrides as it was already taken care of
+                } else {
+                    if (attribute instanceof PropertyAttributeDefinition) {
+                        ((PropertyAttributeDefinition) attribute).setPropertyValue(context, model, configuration);
+                    } else {
+                        final ModelNode value = attribute.resolveModelAttribute(context, model);
+                        if (value.isDefined()) {
+                            if (changeProperty(configuration, attribute.getName(), value.asString())) {
+                                configuration.setPropertyValueString(attribute.getName(), value.asString());
+                            }
+                        } else {
+                            configuration.removeProperty(attribute.getName());
+                        }
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void updateModel(final ModelNode operation, final ModelNode model) throws OperationFailedException {
+            for (AttributeDefinition attribute : ATTRIBUTES) {
+                attribute.validateAndSet(operation, model);
+            }
+        }
+
+        private static boolean changeProperty(final FormatterConfiguration configuration, final String name, final String value) {
```

Comment:
I'm not sure about this method. Given its name, I'd assume it is changing the configuration but it's only checking if the current value is the same. 
Shouldn't it be renamed to something like `isSamePropertyValue()`?